store/src/java/com/zimbra/cs/account/soap/SoapProvisioning.java

```diff
@@ -2938,4 +2938,9 @@ public void autoProvControl(String action) throws ServiceException {
         return result;
     }
 
+    @Override
+    public void refreshUserCredentials(Account account) {
+        // FIXME: Currently no way to refresh required attributes over SOAP
```

Comment:
Hmm.  Tricky.  I'd normally throw a NotImplementedException in these cases.  Tempted to think should do that if this is called things won't work as expected.

Reply:
Fair enough.  At least in this case we don't/can't use SoapProvisioning as a direct swap-in replacement for LdapProvisioning because of this and other 'features' such as the ephemeral data piece. 

Throwing an exception here would make this even more obvious since it would fail if it ever reached this point in the code during runtime.

Will make this update.

Reply:
Do you mean this one? sun.reflect.generics.reflectiveObjects.NotImplementedException;

I am getting a warning telling me it is an internal exception and shouldn't be used. 

     warning: NotImplementedException is internal proprietary API and may be removed in a future release

Reply:
Would adding a new code to ServiceException be the correct course of action?

Reply:
new code for ServiceException is a good idea. As far as I can see, we throw generic exceptions in similar cases elsewhere. So, you could just throw  ServiceException.FAILURE("Currently no way to refresh required attributes over SOAP")

Reply:
Actually - I meant UnsupportedOperationException which is a RuntimeException - that is used in some other places.
Personally, I think ServiceException is a bad idea as it it is treated as something we can deal with in a lot of cases whereas if we're calling this code we're doing something we shouldn't be doing and we need that shouted from the rooftops.

Reply:
Thank you for the update.  updated to throw UnsupportedOperationException